plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/action/GetQueryGroupRequest.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.wlm.action;
+
+import org.opensearch.action.ActionRequest;
+import org.opensearch.action.ActionRequestValidationException;
+import org.opensearch.core.common.io.stream.StreamInput;
+import org.opensearch.core.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+
+/**
+ * Request for get QueryGroup
+ *
+ * @opensearch.experimental
+ */
+public class GetQueryGroupRequest extends ActionRequest {
+    final String name;
+
+    /**
+     * Default constructor for GetQueryGroupRequest
+     * @param name - name for the QueryGroup to get
+     */
+    public GetQueryGroupRequest(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Constructor for GetQueryGroupRequest
+     * @param in - A {@link StreamInput} object
+     */
+    public GetQueryGroupRequest(StreamInput in) throws IOException {
+        super(in);
+        name = in.readOptionalString();
+    }
+
+    @Override
+    public ActionRequestValidationException validate() {
+        return null;
```

Comment:
Do we want to validate the name length, allowed characters before getting the query group from cluster manager?